src/resauce/core.clj

```diff
@@ -12,15 +12,16 @@
     (filter (partial re-matches re) paths)))
 
 (defn- build-url [base-url dir path]
-  (if (.startsWith ^String path dir)
-    (str (add-ending-slash (str base-url))
-         (subs path (count (add-ending-slash dir))))))
+  (assert (.startsWith ^String path dir))
```

Comment:
Why was this changed to an `assert`?

Reply:
Because `build-url` is a private function that's only called in a context where `path` _must_ start with `dir`. Having a condition in there makes it seem like `url-dir` could return a list of `nil`s.

Reply:
Okay. Let's make it a precondition then.